packages/data-stores/src/user/reducer.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { Reducer } from 'redux';
+import { combineReducers } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { ActionType, CurrentUser, NewUser } from './types';
+import * as Actions from './actions';
+
+const currentUser: Reducer<
+	CurrentUser | undefined,
+	| ReturnType< typeof Actions[ 'receiveCurrentUser' ] >
+	| ReturnType< typeof Actions[ 'receiveCurrentUserFailed' ] >
```

Comment:
Doesn't feel like this is going to scale as a reducer takes more actions, but I think this is a pattern that's copied from other data-stores?

Reply:
The thinking with this pattern was to get very detailed information about the actions handled by any given reducer. We're open to feedback and exploration, nothing is set in stone 🙂